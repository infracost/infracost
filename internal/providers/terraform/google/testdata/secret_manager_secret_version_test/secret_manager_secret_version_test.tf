provider "google" {
  credentials = "{\"type\":\"service_account\"}"
  region      = "us-central1"
}

resource "google_secret_manager_secret" "secret_example" {
  secret_id = "secret"

  labels = {
    label = "example"
  }

  replication {
    user_managed {
      replicas {
        location = "us-central1"
      }
      replicas {
        location = "us-east1"
      }
    }
  }
}

resource "google_secret_manager_secret" "secret_automatic" {
  secret_id = "secret-with-usage"

  labels = {
    label = "example-with-usage"
  }

  replication {
    automatic = true
  }
}

resource "google_secret_manager_secret_version" "secret_version" {
  secret = google_secret_manager_secret.secret_example.id

  secret_data = "secret-data"
}

resource "google_secret_manager_secret_version" "secret_version_with_usage" {
  secret = google_secret_manager_secret.secret_automatic.id

  secret_data = "secret-data"
}

resource "google_secret_manager_secret_version" "secret_version_replicas_with_usage" {
  secret = google_secret_manager_secret.secret_example.id

  secret_data = "secret-data"
}
