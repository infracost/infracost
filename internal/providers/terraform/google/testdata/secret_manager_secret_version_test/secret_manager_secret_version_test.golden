
 Name                                                                           Monthly Qty  Unit                    Monthly Cost 
                                                                                                                                  
 google_secret_manager_secret.secret_automatic                                                                                    
 ├─ Active secret versions                                                Monthly cost depends on usage: $0.06 per versions       
 ├─ Access operations                                                     Monthly cost depends on usage: $0.03 per 10K requests   
 └─ Rotation notifications                                                Monthly cost depends on usage: $0.05 per rotations      
                                                                                                                                  
 google_secret_manager_secret.secret_example                                                                                      
 ├─ Active secret versions                                                Monthly cost depends on usage: $0.06 per versions       
 ├─ Access operations                                                     Monthly cost depends on usage: $0.03 per 10K requests   
 └─ Rotation notifications                                                Monthly cost depends on usage: $0.05 per rotations      
                                                                                                                                  
 google_secret_manager_secret_version.secret_version                                                                              
 ├─ Active secret versions                                                                2  versions                       $0.12 
 └─ Access operations                                                     Monthly cost depends on usage: $0.03 per 10K requests   
                                                                                                                                  
 google_secret_manager_secret_version.secret_version_replicas_with_usage                                                          
 ├─ Active secret versions                                                                2  versions                       $0.12 
 └─ Access operations                                                                   0.2  10K requests                   $0.01 
                                                                                                                                  
 google_secret_manager_secret_version.secret_version_with_usage                                                                   
 ├─ Active secret versions                                                                1  versions                       $0.06 
 └─ Access operations                                                                   2.5  10K requests                   $0.08 
                                                                                                                                  
 OVERALL TOTAL                                                                                                              $0.38 
──────────────────────────────────
5 cloud resources were detected:
∙ 5 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file